controllers/metering/controllers/metering_controller.go

```diff
@@ -275,10 +279,151 @@ func NewBillingList(TimeInterval meteringv1.TimeIntervalType, amount int64) mete
 	}
 }
 
+func (r *MeteringReconcile) initMetering(ctx context.Context, ns corev1.Namespace) error {
+	totalResourcePrice, err := r.GetAllExtensionResources(ctx)
+	if err != nil {
+		return err
+	}
+	r.Logger.Info("totalResourcePrice", "totalResourcePrice", totalResourcePrice)
+	if err = r.syncMetering(ctx, ns, totalResourcePrice); err != nil {
+		r.Logger.Error(err, "Failed to create Metering")
+		return client.IgnoreNotFound(err)
+	}
+	return nil
+}
+
+func (r *MeteringReconcile) syncMetering(ctx context.Context, ns corev1.Namespace, resourcePrice map[corev1.ResourceName]meteringv1.ResourcePriceAndUsed) error {
+	metering := meteringv1.Metering{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      MeteringPrefix + ns.Name,
+			Namespace: r.MeteringSystemNameSpace,
+		},
+	}
+
+	r.Logger.V(1).Info("metering env", "METERING_INTERVAL", r.MeteringSystemNameSpace, "timeInterval: ", r.MeteringInterval)
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, &metering, func() error {
+		metering.Spec.Namespace = ns.Name
+		metering.Spec.Owner = ns.Annotations[userv1.UserAnnotationOwnerKey]
+		metering.Spec.Resources = resourcePrice
+		metering.Spec.TimeInterval = r.MeteringInterval
+		return nil
+	}); err != nil {
+		r.Error(err, "Failed to update Metering")
+		return err
+	}
+	r.Logger.Info("metering Spec", "metering.Spec", metering.Spec)
+	// get or create resourceQuota
+	if err := r.syncResourceQuota(ctx, metering); err != nil {
+		r.Error(err, "Failed to syncResourceQuota")
+		return err
+	}
+	return nil
+}
+
+func (r *MeteringReconcile) syncResourceQuota(ctx context.Context, metering meteringv1.Metering) error {
+	quota := &corev1.ResourceQuota{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ResourceQuotaPrefix + metering.Spec.Namespace,
+			Namespace: metering.Spec.Namespace,
+		},
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, quota, func() error {
+		quota.Spec.Hard = meteringv1.DefaultResourceQuota()
+		return nil
+	}); err != nil {
+		return fmt.Errorf("sync resource quota failed: %v", err)
+	}
+	return nil
+}
+
+func (r *MeteringReconcile) DelMetering(ctx context.Context, name, namespace string) error {
```

Comment:
no need to export this func.

Reply:
a sample encapsulation，else need write more code in reconcile